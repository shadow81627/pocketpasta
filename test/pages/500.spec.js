import component from '@/components/error/500';
import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuetify from 'vuetify';

const localVue = createLocalVue();

localVue.use(Vuetify);

const factory = () =>
  shallowMount(component, {
    localVue,
    propsData: {
      error: {
        statusCode: 500,
        message: 'Error',
      },
    },
    stubs: {
      NuxtLink: RouterLinkStub,
    },
  });

describe('500', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    component.$t = (m) => m;
    expect(component.head()).toBeTruthy();
  });
});
