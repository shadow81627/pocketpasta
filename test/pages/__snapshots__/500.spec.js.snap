// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`500 renders properly 1`] = `
"<v-card-stub loaderheight=\\"4\\" tag=\\"div\\" id=\\"message\\">
  <h2>500</h2>
  <h1>error.500.heading</h1>
  <p>
    error.500.description
  </p> <video autoplay=\\"autoplay\\" width=\\"640\\" data-alt=\\"Baboon pounding on laptop keyboard\\" src=\\"https://media.gettyimages.com/videos/medium-shot-baboon-pounding-on-laptop-keyboard-video-id712-54\\" muted=\\"muted\\" loop=\\"loop\\"></video>
  <v-row-stub tag=\\"div\\" nogutters=\\"true\\">
    <v-col-stub tag=\\"div\\">
      <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" href=\\"javascript:location.reload();\\" large=\\"true\\" block=\\"true\\" type=\\"button\\">Try again</v-btn-stub>
    </v-col-stub>
  </v-row-stub>
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\">
      <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" exact=\\"true\\" to=\\"index\\" large=\\"true\\" block=\\"true\\" type=\\"button\\">layout.navigation.home</v-btn-stub>
    </v-col-stub>
  </v-row-stub>
</v-card-stub>"
`;
